### PR TITLE
Call base.Close() in MockFileStream.Close()

### LINKED ISSUE #397: Casting MockFileStream to Stream and then disposing it does not cause subsequent calls to throw ObjectDisposedException
Consider the following (.NET 4.6.2):
```
namespace ConsoleApp1
{
    using System;
    using System.IO;
    using System.IO.Abstractions.TestingHelpers;
    using System.Text;

    class Program
    {
        static void Main(string[] args)
        {
            var stream = File.Create("test.txt");
            RunTest(stream);
            var actualStreamDisposed = TestDisposed(stream);

            var mockFileSystem = new MockFileSystem();
            mockFileSystem.AddDirectory("C:\\");
            var mockStream = mockFileSystem.File.Create("C:\\test.txt");
            RunTest(mockStream);
            var mockStreamDisposed = TestDisposed(mockStream);

            var memoryStream = new MemoryStream();
            RunTest(memoryStream);
            var memoryStreamDisposed = TestDisposed(memoryStream);

            var mockStreamNoCast = (MockFileStream)mockFileSystem.File.Create("C:\\test2.txt");
            RunTestSpecific(mockStreamNoCast);
            var mockStreamSpecificDisposed = TestDisposedSpecific(mockStreamNoCast);

            Console.WriteLine($"Actual stream disposed: {actualStreamDisposed}");
            Console.WriteLine($"Mock stream disposed: {mockStreamDisposed}");
            Console.WriteLine($"Memory stream disposed: {memoryStreamDisposed}");
            Console.WriteLine($"Mock stream (specific) disposed: {memoryStreamDisposed}");
            Console.ReadKey();
        }

        static void RunTest(Stream stream)
        {
            var bytes = Encoding.UTF8.GetBytes("this is only a test");
            stream.Write(bytes, 0, bytes.Length);
            stream.Flush();
            stream.Dispose();
        }

        static void RunTestSpecific(MockFileStream stream)
        {
            var bytes = Encoding.UTF8.GetBytes("this is only a test");
            stream.Write(bytes, 0, bytes.Length);
            stream.Flush();
            stream.Dispose();
        }

        static bool TestDisposed(Stream stream)
        {
            try
            {
                stream.WriteByte(0);
            }
            catch (ObjectDisposedException)
            {
                return true;
            }
            return false;
        }

        static bool TestDisposedSpecific(MockFileStream stream)
        {
            try
            {
                stream.WriteByte(0);
            }
            catch (ObjectDisposedException)
            {
                return true;
            }
            return false;
        }
    }
}
```

Expected output:
```
Actual stream disposed: True
Mock stream disposed: True
Memory stream disposed: True
Mock stream (specific) disposed: True
```

Actual output:
```
Actual stream disposed: True
Mock stream disposed: False
Memory stream disposed: True
Mock stream (specific) disposed: True
```

I included `MemoryStream` because I saw that `MockFileStream` inherits from `MemoryStream` so I wanted to make sure that it wasn't actually a problem in `MemoryStream`.

If you don't cast to `Stream`, it works as expected.

The issue is seems to be that `Close` in `MockFileStream` doesn't call through to `base.Close()` because if I add that, it works as expected.

### PR DESCRIPTION
**Pull request by biltongza**
Fixes #397.

### CODE REVIEW

File: System.IO.Abstractions.TestingHelpers/MockFileStream.cs [hunk 1]
```diff
@@ -64,8 +65,14 @@ public enum StreamType
 #if NET40
         public override void Close()
         {
+            if(closed)
```
[the remark this answers is not in the document]
Comment (biltongza) [reply System.IO.Abstractions.TestingHelpers/MockFileStream.cs]: Fixed.

**Comment by biltongza** [comment]
Cool! Any chance of getting a beta with this in it? I need this to fix a failing test at work :)

**Comment by robkeim** [comment]
@biltongza I'll defer to some of the main contributors on that @fgreinacher and @jpreese can you guys weigh in?

**Comment by fgreinacher** [comment]
LGTM

**Comment by fgreinacher** [comment]
@biltongza Will be available at https://www.nuget.org/packages/System.IO.Abstractions.TestingHelpers/2.2.8-beta in some minutes.

**Pull request merged**